src/oscar/apps/partner/abstract_models.py

```diff
@@ -304,3 +308,53 @@ class Meta:
         ordering = ('-date_created',)
         verbose_name = _('Stock alert')
         verbose_name_plural = _('Stock alerts')
+
+
+@python_2_unicode_compatible
+class AbstractLegalEntity(models.Model):
+    """
+    Represents LegalEntity - merchant (company or individual) which we issue
+    invoice on behalf of.
+    """
+    shop_name = models.CharField(_('Platform name'), max_length=255, null=True, blank=True)
+    business_name = models.CharField(_('Business name'), max_length=255, db_index=True)
+    vat_number = models.CharField(_('VAT identification number'), max_length=20)
+    logo = models.ImageField(
+        _('Logo'), upload_to=settings.OSCAR_IMAGE_FOLDER, max_length=255, null=True, blank=True)
+    email = models.EmailField(_('Email'), null=True, blank=True)
+    web_site = models.URLField(_('Website'), null=True, blank=True)
+
+    class Meta:
+        abstract = True
+        app_label = 'partner'
+        verbose_name = _('Legal Entity')
+        verbose_name_plural = _('Legal Entities')
+
+    def __str__(self):
+        return self.company_name
+
+    @property
+    def has_addresses(self):
+        return self.addresses.exists()
+
+
+class AbstractLegalEntityAddress(AbstractAddress):
+    """
+    Represents Address of LegalEntity.
+
+    Used in Invoices.
+    """
+    legal_entity = models.ForeignKey(
+        'partner.LegalEntity',
+        on_delete=models.CASCADE,
+        related_name='addresses',
+        verbose_name=_('Legal Entity'))
+
+    phone_number = PhoneNumberField(_('Phone number'), null=True, blank=True)
```

Comment:
phone_number and fax_number shouldn't bei null=True because the PhoneNumberField is based upon CharField.
See https://github.com/stefanfoulis/django-phonenumber-field/blob/master/README.rst